pkg/logging/digitalocean/digitalocean_test.go

```diff
@@ -0,0 +1,489 @@
+		Access key: 1234
+		Secret key: -----BEGIN RSA PRIVATE KEY-----MIIEogIBAAKCA
+		Path: logs/
+		Period: 86400
+		GZip level: 9
+		Format: %h %l %u %t "%r" %>s %b
+		Format version: 2
+		Response condition: Prevent default logging
+		Message type: classic
+		Timestamp format: %Y-%m-%dT%H:%M:%S.000
+		Placement: none
+		Public key: `+pgpPublicKey()+`
+`) + "\n\n"
+
+func getDigitalOceanOK(i *fastly.GetDigitalOceanInput) (*fastly.DigitalOcean, error) {
+	return &fastly.DigitalOcean{
+		ServiceID:         i.Service,
+		Version:           i.Version,
+		Name:              "logs",
+		BucketName:        "my-logs",
+		Domain:            "https://digitalocean.us-east-1.amazonaws.com",
+		AccessKey:         "1234",
+		SecretKey:         "-----BEGIN RSA PRIVATE KEY-----MIIEogIBAAKCA",
+		Path:              "logs/",
+		Period:            3600,
+		GzipLevel:         9,
+		Format:            `%h %l %u %t "%r" %>s %b`,
+		FormatVersion:     2,
+		ResponseCondition: "Prevent default logging",
+		MessageType:       "classic",
+		TimestampFormat:   "%Y-%m-%dT%H:%M:%S.000",
+		Placement:         "none",
+		PublicKey:         pgpPublicKey(),
+	}, nil
+}
+
+func getDigitalOceanError(i *fastly.GetDigitalOceanInput) (*fastly.DigitalOcean, error) {
+	return nil, errTest
+}
+
+var describeDigitalOceanOutput = strings.TrimSpace(`
+Service ID: 123
+Version: 1
+Name: logs
+Bucket: my-logs
+Domain: https://digitalocean.us-east-1.amazonaws.com
+Access key: 1234
+Secret key: -----BEGIN RSA PRIVATE KEY-----MIIEogIBAAKCA
+Path: logs/
+Period: 3600
+GZip level: 9
+Format: %h %l %u %t "%r" %>s %b
+Format version: 2
+Response condition: Prevent default logging
+Message type: classic
+Timestamp format: %Y-%m-%dT%H:%M:%S.000
+Placement: none
+Public key: `+pgpPublicKey()+`
+`) + "\n"
+
+func updateDigitalOceanOK(i *fastly.UpdateDigitalOceanInput) (*fastly.DigitalOcean, error) {
+	return &fastly.DigitalOcean{
+		ServiceID:         i.Service,
+		Version:           i.Version,
+		Name:              "log",
+		BucketName:        "my-logs",
+		Domain:            "https://digitalocean.us-east-1.amazonaws.com",
+		AccessKey:         "1234",
+		SecretKey:         "-----BEGIN RSA PRIVATE KEY-----MIIEogIBAAKCA",
+		Path:              "logs/",
+		Period:            3600,
+		GzipLevel:         9,
+		Format:            `%h %l %u %t "%r" %>s %b`,
+		FormatVersion:     2,
+		ResponseCondition: "Prevent default logging",
+		MessageType:       "classic",
+		TimestampFormat:   "%Y-%m-%dT%H:%M:%S.000",
+		Placement:         "none",
+		PublicKey:         pgpPublicKey(),
+	}, nil
+}
+
+func updateDigitalOceanError(i *fastly.UpdateDigitalOceanInput) (*fastly.DigitalOcean, error) {
+	return nil, errTest
+}
+
+func deleteDigitalOceanOK(i *fastly.DeleteDigitalOceanInput) error {
+	return nil
+}
+
+func deleteDigitalOceanError(i *fastly.DeleteDigitalOceanInput) error {
+	return errTest
+}
+
+// pgpPublicKey returns a PEM encoded PGP public key suitable for testing.
+func pgpPublicKey() string {
```

Comment:
This is repeated in quite a few endpoints. However, I don't think that we should create an exported function for it in testutil.